Hoarder enhancement

vhsdream opened the pull request:
This PR offers a successful build of Hoarder, with essential optional dependencies included (Chromium, Monolith, Meilisearch). I've run it a few times on my node and it seems to build ok.

A few things:

- This has 4 systemd service files instead of 1 - for the browser, for the workers, for the web frontend, and a .target file so it is easier to start the app.
- There is also a meilisearch service file, that is not a part of the above.
- I messed with the paths in the /opt/hoarder folder. I bet that was not the greatest idea, and that could probably be optimized.
- I have not changed anything in the ct/hoarder.sh update_script function besides removing the old space and resources check and replacing with the new options; the actual updating part of it is not going to work now.
- After the successful build, the RAM gets reduced to 2GB - I found that 2GB build would fail during the web frontend build phase so 4 is a good number for that.

MohamedBassem commented:
Oh lol, only saw this PR after sending mine (https://github.com/MickLesk/Proxmox_DEV/pull/3/files) :)

MohamedBassem commented:
Worth noting that some of the installation steps here are a bit more complicated than they need to be. They are done that way in the docker container to reduce the image size, though this is typically not a big concern in LXCs installations.
For example, you don't need to `pnpm deploy` the workers container, or run `ncc` on the db migration dir. We can just use the main installation directory for all of that (similar to that of my PR).

As I mentioned in the other discussion, I think the final script is going to be a mix of both PRs.

vhsdream commented:
Hi, so I've added the changes from @MohamedBassem's PR to mine, and tested it extensively. I added the database migration command near the end, before starting the services. I've also left in the Meilisearch, Chromium, Monolith and yt-dlp dependencies; but perhaps it might be worth it to make yt-dlp and Monolith as togglable options in the UI in case people do not want to archive entire webpage ir download videos.
